backend/remote/backend_common.go

```diff
@@ -228,37 +228,51 @@ func (b *Remote) parseVariableValues(op *backend.Operation) (terraform.InputValu
 }
 
 func (b *Remote) costEstimation(stopCtx, cancelCtx context.Context, op *backend.Operation, r *tfe.Run) error {
+	if r.CostEstimation != nil {
+		return nil
+	}
+
 	if b.CLI != nil {
 		b.CLI.Output("\n------------------------------------------------------------------------\n")
 	}
-	var ce = r.CostEstimation
 
-	logs, err := b.client.CostEstimations.Logs(stopCtx, ce.ID)
+	logs, err := b.client.CostEstimations.Logs(stopCtx, r.CostEstimation.ID)
 	if err != nil {
 		return generalError("Failed to retrieve cost estimation logs", err)
 	}
-	scanner := bufio.NewScanner(logs)
+	reader := bufio.NewReaderSize(logs, 64*1024)
 
 	// Retrieve the cost estimation to get its current status.
-	ce, err = b.client.CostEstimations.Read(stopCtx, ce.ID)
+	ce, err := b.client.CostEstimations.Read(stopCtx, r.CostEstimation.ID)
 	if err != nil {
 		return generalError("Failed to retrieve cost estimation", err)
 	}
 
-	var msgPrefix = "Cost estimation"
-
+	msgPrefix := "Cost estimation"
 	if b.CLI != nil {
 		b.CLI.Output(b.Colorize().Color(msgPrefix + ":\n"))
 	}
 
-	for scanner.Scan() {
-		if b.CLI != nil {
-			b.CLI.Output(b.Colorize().Color(scanner.Text()))
+	if b.CLI != nil {
```

Comment:
Are we really expecting the cost estimation to have lines longer than 64K?

Reply:
No, but I like to keep the log reading logic consistent and the format may change in the future. So better safe than sorry right?

Reply:
From a maintenance standpoint, the next person in here is probably going to wonder why this didn't just use `scanner.Scan` and save the 10 lines of finicky logic. 

If you're concerned that you may have lines over 64k at some point, then we can keep it, otherwise I think it's a lot of unneeded surface area.

Reply:
@thrashr888 I guess you can answer the questions about what we can/should expect? I'm fine with reverting this to using a scanner.

Reply:
We should not run into 64k lines but we do print out resource names and I'm not sure of their max-length. I can switch it back.